core/container/mutexMap_test.go

```diff
@@ -0,0 +1,265 @@
+	valToStore := 56
+	mm.Set(key, valToStore)
+
+	val, ok = mm.Get(key)
+	assert.Equal(t, valToStore, val)
+	assert.True(t, ok)
+}
+
+func TestMutexMap_GetConcurrent(t *testing.T) {
+	t.Parallel()
+
+	mm := NewMutexMap()
+	key := "existent key"
+	valToStore := 56
+	mm.Set(key, valToStore)
+
+	numIterations := 100
+	wg := &sync.WaitGroup{}
+	wg.Add(numIterations)
+	for i := 0; i < numIterations; i++ {
+		go func() {
+			val, ok := mm.Get(key)
+			assert.Equal(t, valToStore, val)
+			assert.True(t, ok)
+
+			wg.Done()
+		}()
+	}
+
+	wg.Wait()
+}
+
+//------- Set
+
+func TestMutexMap_SetConcurrent(t *testing.T) {
+	t.Parallel()
+
+	mm := NewMutexMap()
+	key := "existent key"
+	valToStore := 56
+
+	numIterations := 100
+	wg := &sync.WaitGroup{}
+	wg.Add(numIterations)
+	for i := 0; i < numIterations; i++ {
+		go func() {
+			mm.Set(key, valToStore)
+			wg.Done()
+		}()
+	}
+
+	wg.Wait()
+
+	val, ok := mm.Get(key)
+	assert.Equal(t, valToStore, val)
+	assert.True(t, ok)
+}
+
+//------- Insert
+
+func TestMutexMap_Insert(t *testing.T) {
+	t.Parallel()
+
+	mm := NewMutexMap()
+
+	key := "key"
+	val1 := 56
+	val2 := 78
+
+	//operation succeed
+	ok := mm.Insert(key, val1)
+	assert.True(t, ok)
+
+	//test inner value
+	recoveredValue, _ := mm.Get(key)
+	assert.Equal(t, val1, recoveredValue)
+
+	//rewrite will fail
+	ok = mm.Insert(key, val2)
+	assert.False(t, ok)
+
+	//test inner value
+	recoveredValue, _ = mm.Get(key)
+	assert.Equal(t, val1, recoveredValue)
+}
+
+func TestMutexMap_InsertConcurrent(t *testing.T) {
+	t.Parallel()
+
+	mm := NewMutexMap()
+
+	numIterations := 100
+	wg := &sync.WaitGroup{}
+	wg.Add(numIterations)
+	for i := 0; i < numIterations; i++ {
+		go func(idx int) {
+			key := fmt.Sprintf("%d", idx)
+			ok := mm.Insert(key, struct{}{})
+			assert.True(t, ok)
+
+			wg.Done()
+		}(i)
+	}
+
+	wg.Wait()
+}
+
+//------- Remove
+
+func TestMutexMap_Remove(t *testing.T) {
+	t.Parallel()
+
+	mm := NewMutexMap()
+
+	key := "key"
+	mm.Remove(key)
+
+	mm.Set(key, struct{}{})
+	valRecovered, _ := mm.Get(key)
+	assert.NotNil(t, valRecovered)
+
+	mm.Remove(key)
+
+	valRecovered, _ = mm.Get(key)
+	assert.Nil(t, valRecovered)
+}
+
+func TestMutexMap_RemoveConcurrentShouldNotPanic(t *testing.T) {
+	t.Parallel()
+
+	defer func() {
+		r := recover()
+		if r != nil {
+			assert.Fail(t, fmt.Sprintf("should not have paniced %v", r))
+		}
+	}()
+
+	mm := NewMutexMap()
+
+	numIterations := 100
+	wg := &sync.WaitGroup{}
+	wg.Add(numIterations)
+	for i := 0; i < numIterations; i++ {
+		go func(idx int) {
+			mm.Remove("key")
+
+			wg.Done()
+		}(i)
+	}
+
+	wg.Wait()
+}
+
+//------- Keys
+
+func TestMutexMap_Keys(t *testing.T) {
+	t.Parallel()
+
+	mm := NewMutexMap()
+
+	key1 := "key1"
+	key2 := "key2"
+	key3 := "key3"
+
+	keys := mm.Keys()
+	assert.Equal(t, 0, len(keys))
+
+	mm.Set(key1, struct{}{})
+	mm.Set(key2, struct{}{})
+	mm.Set(key3, struct{}{})
+
+	keys = mm.Keys()
+	require.Equal(t, 3, len(keys))
+	stringKeys := make([]string, len(keys))
```

Comment:
From this line below, you only test for the existence of the keys inside the `mm.Keys()` slice. Sorting the keys and then performing the assertions in order does not improve the test (length assertion performed, and keys are unique). Is this correct? `require.Contains()` can be an alternative.